### PR TITLE
ug update

### PR DESCRIPTION
**Pull request by Thing1Thing2**
Changes made:
- Add user interface description
- Add to troubleshooting
- Add acknowledgement section
- Add `clear` feature
- Update FAQ 
- Update glossary
- Renumber sections
- Check hyperlinks
- Move explanation of `INDEX` to glossary (to avoid repetition in each command)
- Standardize italicizing of "MyInsuRec" to *MyInsuRec*

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -90,7 +81,20 @@ Beyond individual products, you can also:
...
+
+The main window shows the list of clients, meetings or products on executing a list command.
+It also shows a lists of details associated with a client, meeting, or product on executing a view command.
+
+All commands are accepted from the user via the [CLI](#cli).
+
+The result of executing a command is shown in the command result window just above the CLI.
+Here you will see help, command success, and warning messages.
+
+![Ui](images/annotatedGui.png)
```
Comment (rexong) [inline docs/UserGuide.md:97]: Hi I think is portion should be done by me. Not sure why you had this filled in as well

File: docs/UserGuide.md [hunk 2]
```diff
@@ -270,22 +273,19 @@ View details associated with a client, such as the client's name and phone numbe
...
+* Display information about the client at the specific [`INDEX`](#index) in the client list.
+
 
 Examples:
 * `viewClient i/1`
 
 #### 5.1.4 Delete client: `delClient`
 
-Delete the specified client from MyInsuRec.
+Delete the specified client from *MyInsuRec*.
```
Comment (rexong) [inline docs/UserGuide.md:284]: `*MyInsuRec*` should probably be change to `_MyInsuRec_`

Comment (rexong) [reply docs/UserGuide.md:284]: Just wanna check is it in italic? and also why are the 2 asterisks at the start and end?

File: docs/UserGuide.md [hunk 3]
```diff
@@ -441,24 +431,47 @@ Show a message explaining how to access the help page.
...
 
 <div markdown="span" class="alert alert-warning">:exclamation: **Caution:**
-If your changes to the data file makes its format invalid, MyInsuRec will discard all data and start with an empty data file at the next run.
+If your changes to the data file makes its format invalid, *MyInsuRec* will discard all data and start with an empty data file at the next run.
+</div>
+
+#### 5.5.3 Creating and using data file backups
+
+This is the recommended method of storing backups of the `[JAR file location]/data/myinsurec.json` file.
+First, create a copy of the original file in the `[JAR file location]/data` folder and rename it to `myinsurec_date.json` where date identifies when the file was created.
```
Comment (rexong) [inline docs/UserGuide.md:468]: for the `date` here are you refering to `myinsurec_29102022`?

File: docs/UserGuide.md [hunk 4]
```diff
@@ -524,8 +546,106 @@ To check that your system has the correct Java version (Java 11 and above) to ru
...
+Do not launch the app if you detect this issue, since the `myinsurec.json` file will be cleared of all data.
+To recover:
+
+1. [create a backup](#553-creating-and-using-data-file-backups) of the file.
+2. Edit all invalid fields and format issues manually in the backup file.
+    * What should I look out for while editing? :
+    * Check that the structure of the file is correct (no missing semicolons, commas, brackets, and data fields). <br> It helps to be familiar with the [JSON file](#json-file) format here, but most users can quickly pick up the basics by referring to a working copy of the data file.
+    * Some invalid data values to look out for :
+    * invalid dates for client birthday, or meeting date e.g. 00000000 (invalid DDMMYYYY), 1212202020 (too many digits),
+    * Invalid meeting start time or end time e.g. 2500 (invalid HHMM).
```
Comment (rexong) [inline docs/UserGuide.md:576]: should have these with more one indentation.

File: docs/UserGuide.md [hunk 5]
```diff
@@ -524,8 +546,106 @@ To check that your system has the correct Java version (Java 11 and above) to ru
...
+    * [Swap the file with your backup](#553-creating-and-using-data-file-backups).
+2. The data file was corrupt before starting the app:
+    * This issue is harder to detect since the app will only empty the file if the data in it was corrupt before starting.
+    * How to avoid this issue? :
+    * [Create backups](#553-creating-and-using-data-file-backups) of your `myinsurec.json` file and ensure that they are not corrupt.
+    * See the [section below](#83-my-data-file-is-corrupt) on how to detect and recover from file corruption.
+
+### 8.3 My data file is corrupt
+
+Do not launch the app if you detect this issue, since the `myinsurec.json` file will be cleared of all data.
```
Comment (rexong) [inline docs/UserGuide.md:567]: How can they detect this issue?

File: docs/UserGuide.md [hunk 6]
```diff
@@ -524,8 +546,106 @@ To check that your system has the correct Java version (Java 11 and above) to ru
...
+1. The data file is empty:
+    * Locate the [date file](#6-faq).
+    * If the file is empty ,i.e. contains no data, then *MyInsuRec* did not load any data on start up.
+    * How to recover? :
+    * [Swap the file with your backup](#553-creating-and-using-data-file-backups).
+2. The data file was corrupt before starting the app:
+    * This issue is harder to detect since the app will only empty the file if the data in it was corrupt before starting.
+    * How to avoid this issue? :
+    * [Create backups](#553-creating-and-using-data-file-backups) of your `myinsurec.json` file and ensure that they are not corrupt.
+    * See the [section below](#83-my-data-file-is-corrupt) on how to detect and recover from file corruption.
```
Comment (rexong) [inline docs/UserGuide.md:563]: Should have another level of indentation

File: docs/UserGuide.md [hunk 7]
```diff
@@ -524,8 +546,106 @@ To check that your system has the correct Java version (Java 11 and above) to ru
...
+
+### 8.2 I cannot see any data
+
+Here are a few reasons why this may have occurred:
+
+1. The data file is empty:
+    * Locate the [date file](#6-faq).
+    * If the file is empty ,i.e. contains no data, then *MyInsuRec* did not load any data on start up.
+    * How to recover? :
+    * [Swap the file with your backup](#553-creating-and-using-data-file-backups).
```
Comment (rexong) [inline docs/UserGuide.md:558]: Should have another level of indentation

File: docs/UserGuide.md [hunk 8]
```diff
@@ -524,8 +546,106 @@ To check that your system has the correct Java version (Java 11 and above) to ru
...
+In `delProduct` command, `INDEX` refers to the number shown by executing [`listProduct`](#532-list-products-listproduct) command. 
+
+e.g. Product6 has position of 5 in the shown [product list](#332-listing-products-listproduct):
+![](images/listProduct.png)
+
+### *J*
+
+##### JSON file
+
+A file that stores data that has been structured according to the JSON data format.
```
Comment (rexong) [inline docs/UserGuide.md:638]: Could proabably provide a link to some online resources to know more about json format

**Review by rexong — CHANGES_REQUESTED** [review]

File: docs/UserGuide.md (hunk 2 above)
Comment (sikai00) [reply docs/UserGuide.md:284]: Actually this is italics, both works fine

File: docs/UserGuide.md (hunk 1 above)
Comment (Thing1Thing2) [reply docs/UserGuide.md:97]: my bad, didn't check that it was assigned already 😅

File: docs/UserGuide.md (hunk 3 above)
Comment (Thing1Thing2) [reply docs/UserGuide.md:468]: yes, its really up to the user how they want to name their files but I thought giving suggestion might help. I'll add the example you have to the guide.

File: docs/UserGuide.md (hunk 4 above)
Comment (Thing1Thing2) [reply docs/UserGuide.md:576]: ok :+1:

File: docs/UserGuide.md (hunk 6 above)
Comment (Thing1Thing2) [reply docs/UserGuide.md:563]: ok :+1:

File: docs/UserGuide.md (hunk 8 above)
Comment (Thing1Thing2) [reply docs/UserGuide.md:638]: ahh good idea, will add that in

File: docs/UserGuide.md (hunk 3 above)
Comment (rexong) [reply docs/UserGuide.md:468]: would prefer a format like `myinsurec_DATE.json` where `DATE` is the date where the file is created or sth like that

File: docs/UserGuide.md (hunk 5 above)
Comment (rexong) [reply docs/UserGuide.md:567]: Hmm, I'm not too sure how to explain this, would probably like to get feedback from the rest!

File: docs/UserGuide.md [hunk 9]
```diff
@@ -501,7 +491,7 @@ If your changes to the data file makes its format invalid, *MyInsuRec* will disc
 #### 5.5.3 Creating and using data file backups
 
 This is the recommended method of storing backups of the `[JAR file location]/data/myinsurec.json` file.
-First, create a copy of the original file in the `[JAR file location]/data` folder and rename it to `myinsurec_date.json` where date identifies when the file was created.
+First, create a copy of the original file in the `[JAR file location]/data` folder and rename it to `myinsurec_date.json` where date identifies when the file was created (e.g. `myinsurec_01122022` for a file created on 1st December 2022).
```
Comment (rexong) [inline docs/UserGuide.md:494]: just a personally perference to have `myinsurec_date.json` as `myinsurec_DATE.json`

Comment (Thing1Thing2) [reply docs/UserGuide.md:494]: okk

File: docs/UserGuide.md [hunk 10]
```diff
@@ -145,28 +135,28 @@ Both client name and their phone number are **needed** to add the client into *M
 
 ![EditClientTommy](images/quick-start/EditClientTommy.png)
 
-4. We can also directly add these optional fields (email, address, etc.) when we are adding the client for the first time! So, `addClient n/Tammy Lim p/90127654 e/tammylim@hotmail.com b/09091973` also adds in the client's email and birthday directly.
+4. We can also directly add these optional fields (email, address, etc.) when we are adding the client for the first time! So, `addClient n/Tammy Lim p/90127654 e/tammylim@hotmail.com b/09091973` also adds in the client's email and birthday directly.
```
Comment (ThomasHoooo) [inline docs/UserGuide.md:138]: extra space

File: docs/UserGuide.md [hunk 11]
```diff
@@ -481,24 +457,47 @@ Show a message explaining how to access the help page.
...
+
+Format: `clear`
+
+This is a one-shot way of removing all data stored in *MyInsuRec*.
+Only clear when you are sure you do not need the data anymore, or else make a [backup](#553-creating-and-using-data-file-backups).
+
+Example usage: 
+* Clearing default data shipped with *MyInsuRec*
+* Restarting the app from an empty version
+
```
Comment (ThomasHoooo) [inline docs/UserGuide.md:476]: update to the new format

File: docs/UserGuide.md [hunk 12]
```diff
@@ -481,24 +457,47 @@ Show a message explaining how to access the help page.
...
+This is the recommended method of storing backups of the `[JAR file location]/data/myinsurec.json` file.
+First, create a copy of the original file in the `[JAR file location]/data` folder and rename it to `myinsurec_DATE.json` where date identifies when the file was created (e.g. `myinsurec_01122022` for a file created on 1st December 2022).
+
+If you want to start *MyInsuRec* using a backup file, rename the backup to `myinsurec.json`.
+Then launch the app as usual.
 
 <div markdown="span" class="alert alert-warning">:exclamation: **Caution:**
-If your changes to the data file makes its format invalid, MyInsuRec will discard all data and start with an empty data file at the next run.
+Remember to save the original `myinsurec.json` file before renaming the backup!
```
Comment (ThomasHoooo) [inline docs/UserGuide.md:500]: should be bold instead of backticks

**Pull request merged**